EnrichStream/Dockerfile

```diff
@@ -1,58 +1,38 @@
 #
-#  Install dependencies, compile sources, build jars.
+#  Install dependencies, compile sources, build jar.
 #
 FROM maven:3.9.6-eclipse-temurin-17-alpine as build
 
 WORKDIR /app
 
-ARG MAVEN_ARGS
-
 COPY pom.xml ./
 COPY settings.xml ./
-# Copy in local repository
-COPY .m2/ /root/.m2/
+COPY .mvn .mvn
 
 # Download dependencies (with docker cache)
-RUN \
-    --mount=type=secret,id=GITHUB_TOKEN \
+RUN --mount=type=secret,id=GITHUB_TOKEN \
     --mount=type=secret,id=GITHUB_ACTOR \
-    --mount=type=cache,target=./.m2,sharing=locked \
-    cp -rn /root/.m2/ ./ &&\
+    --mount=type=cache,target=/root/.m2,sharing=locked \
     GITHUB_TOKEN="$(cat /run/secrets/GITHUB_TOKEN)" \
     GITHUB_ACTOR="$(cat /run/secrets/GITHUB_ACTOR)" \
-    mvn dependency:resolve dependency:resolve-plugins -B -U -s ./settings.xml -Dmaven.repo.local="${CI_PROJECT_DIR-.}"/.m2/repository
+    mvn dependency:go-offline -B
 
 # Then add source.
 COPY src ./src
 
 # Finally compile, test, and jar up
-RUN \
-    --mount=type=secret,id=GITHUB_TOKEN \
+RUN --mount=type=secret,id=GITHUB_TOKEN \
     --mount=type=secret,id=GITHUB_ACTOR \
-    --mount=type=cache,target=./.m2,sharing=locked \
+    --mount=type=cache,target=/root/.m2,sharing=locked \
     GITHUB_TOKEN="$(cat /run/secrets/GITHUB_TOKEN)" \
     GITHUB_ACTOR="$(cat /run/secrets/GITHUB_ACTOR)" \
-    mvn package -B -s ./settings.xml ${MAVEN_ARGS-} -Dmaven.repo.local="${CI_PROJECT_DIR-.}"/.m2/repository
+    mvn package -B
 
 #
-#  Install jars onto clean runtime system.
+#  Install jar onto clean runtime system.
 #
-FROM flink:1.18.1-scala_2.12-java17 as runtime
+FROM ghcr.io/trifork/cheetah-lib-processing:4.2.0-feature-lib-processing-image-snapshot36-flink-1.18.1-java17
```

Comment:
should it reference a snapshot?

Reply:
yes, i would like to merge example-flink first using the lib-processing snapshot and then lib-processing e2e tests will go through since they use the main branch of example-flink